tracer/test/Datadog.Trace.Tests/Util/UnmanagedMemoryPoolTests.cs

```diff
@@ -94,14 +94,17 @@ public void TestWithEncoder(int? poolSize)
 
         var pool = Encoder.Pool;
 
-        try
+        for (int x = 0; x < 10; x++)
```

Comment:
We don't seem to have any tests for the behaviour when the number of memory pools exceeds the threshold?